src/traces/parcoords/defaults.js

```diff
@@ -51,29 +51,41 @@ function dimensionsDefaults(traceIn, traceOut) {
             continue;
         }
 
-        coerce('label');
-        coerce('tickvals');
-        coerce('ticktext');
-        coerce('visible');
-        coerce('range');
-        coerce('constraintrange');
-        coerce('values');
+        var userVisible = coerce('visible');
 
         // turn dimensions with no data invisible
-        dimensionOut.visible = dimensionOut.visible && dimensionOut.values.length > 0;
+        var actuallyVisible = userVisible && Lib.isArray(dimensionIn.values) && dimensionIn.values.length > 0;
+
+        if(actuallyVisible) {
```

Comment:
I'd think that this combination:
```
+        var values = coerce('values');
+        var visible = coerce('visible', values.length > 0);
```
has the effect that `values` gets coerced even if the dimension will ultimately be invisible. For example, `values` might not be supplied, or supplied as `null` or `"Hello World"` etc. I think you said that if it's invisible, then the other attributes shouldn't be coerced to anything; it should just be a pass-through. But as I said above, I might have misunderstood this aspect.